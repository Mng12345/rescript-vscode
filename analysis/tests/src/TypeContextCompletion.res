type someVariant = One | Two | Three | Four

let someVariantToString = (~someVariant) =>
  switch someVariant {
  | One => "One"
  | Two => "Two"
  | Three => "Three"
  | Four => "Four"
  }

module SomeComponent = {
  @react.component
  let make = (~whatever) => {
    someVariantToString(~someVariant=whatever)->React.string
  }
}

// let x = someVariantToString(~someVaria
//                                       ^com

// let x = someVariantToString(~someVariant=
//                                          ^com

// let x = someVariantToString(~someVariant=
//                                           ^com

// let x = someVariantToString(~someVariant=T
//                                           ^com

// let x = someVariantToString(~someVariant=t
//                                           ^com

// let jsx = <SomeComponent whatever=
//                                   ^com
