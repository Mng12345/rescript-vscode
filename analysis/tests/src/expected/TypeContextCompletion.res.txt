Complete src/TypeContextCompletion.res 10:41
posCursor:[10:41] posNoWhite:[10:40] Found expr:[10:11->24:1]
Pexp_apply ...[10:11->10:30] (~someVaria10:32->10:41=...[10:32->10:41], ...[19:0->23:3])
Completable: CnamedArg(Value[someVariantToString], someVaria, [someVaria])
Found type for function (~someVariant: someVariant) => string
[{
    "label": "someVariant",
    "kind": 4,
    "tags": [],
    "detail": "someVariant",
    "documentation": null
  }]

Complete src/TypeContextCompletion.res 13:44
posCursor:[13:44] posNoWhite:[13:43] Found expr:[13:11->24:1]
Pexp_apply ...[13:11->13:30] (~someVariant13:32->13:43=...[19:0->23:3])
found typed context 
Completable: CtypedContext(Value[someVariantToString])
[]

Complete src/TypeContextCompletion.res 16:45
posCursor:[16:45] posNoWhite:[16:44] Found expr:[16:11->24:1]
Pexp_apply ...[16:11->16:30] (~someVariant16:32->16:43=...[16:44->16:45], ...[19:0->23:3])
found typed context 
Completable: CtypedContext(Value[someVariantToString])
[]

Complete src/TypeContextCompletion.res 26:37
posCursor:[26:37] posNoWhite:[26:36] Found expr:[26:14->26:37]
JSX <SomeComponent:[26:14->26:27] whatever[26:28->26:36]=...__ghost__[0:-1->0:-1]> _children:None
[]

